drydock_provisioner/drivers/oob/pyghmi_driver/__init__.py

```diff
@@ -172,115 +172,133 @@ def execute_task(self):
 
             i = 18
 
             while i > 0:
-                power_state = ipmi_session.get_power()
+                self.logger.debug("Polling powerstate waiting for success.")
+                power_state = self.exec_ipmi_command(Command.get_power)
                 if power_state.get('powerstate', '') == 'off':
+                    self.logger.debug("Node reports powerstate of off")
+                    worked = True
                     break
                 time.sleep(10)
                 i = i - 1
 
-            if power_state.get('powerstate', '') == 'off':
+            if worked:
                 self.orchestrator.task_field_update(self.task.get_id(),
                     result=hd_fields.ActionResult.Success,
                     status=hd_fields.TaskStatus.Complete)
             else:
+                self.logger.error("Giving up on IPMI command to %s" % self.node.name)
                 self.orchestrator.task_field_update(self.task.get_id(),
                     result=hd_fields.ActionResult.Failure,
                     status=hd_fields.TaskStatus.Complete)
             return
         elif task_action == hd_fields.OrchestratorAction.PowerOnNode:
-            ipmi_session.set_power('on')
+            worked = False
+
+            self.logger.debug("Sending set_power = off command to %s" % self.node.name)
+            self.exec_ipmi_command(Command.set_power, 'off')
 
             i = 18
 
             while i > 0:
-                power_state = ipmi_session.get_power()
-                if power_state.get('powerstate', '') == 'on':
+                self.logger.debug("Polling powerstate waiting for success.")
+                power_state = self.exec_ipmi_command(Command.get_power)
+                if power_state.get('powerstate', '') == 'off':
+                    self.logger.debug("Node reports powerstate of off")
+                    worked = True
                     break
                 time.sleep(10)
                 i = i - 1
 
-            if power_state.get('powerstate', '') == 'on':
+            if worked:
                 self.orchestrator.task_field_update(self.task.get_id(),
                     result=hd_fields.ActionResult.Success,
                     status=hd_fields.TaskStatus.Complete)
             else:
+                self.logger.error("Giving up on IPMI command to %s" % self.node.name)
                 self.orchestrator.task_field_update(self.task.get_id(),
                     result=hd_fields.ActionResult.Failure,
                     status=hd_fields.TaskStatus.Complete)
             return
         elif task_action == hd_fields.OrchestratorAction.PowerCycleNode:
-            ipmi_session.set_power('off')
+            self.logger.debug("Sending set_power = off command to %s" % self.node.name)
+            self.exec_ipmi_command(Command.set_power, 'off')
 
             # Wait for power state of off before booting back up
             # We'll wait for up to 3 minutes to power off
             i = 18
```

Comment:
It would be better if this were controlled by a config option. Same thing with the sleep of 10 seconds.